window/src/lib.rs

```diff
@@ -270,6 +270,14 @@ pub trait WindowOps {
     /// click before forwarding the event (Windows).
     fn set_window_drag_position(&self, _coords: ScreenPoint) {}
 
+    /// Returns the location of the window on the screen.
+    /// The coordinates are of the top left pixel of the
+    /// client area.
+    ///
+    /// This is only implemented on backends that allow
+    /// windows to move themselves (not Wayland).
+    fn get_window_position(&self) -> Future<ScreenPoint>;
```

Comment:
I wonder, should this be `get_window_geometry` instead, and return a `ScreenRect`; since the implementation on all platforms will have the window rectangle, it feels like it might simplify some cases for callers in the future to return it all, especially because right now the window size is tracked in TermWindow based on WindowEvents.

Reply:
That way, the wayland implementation could return the dimensions but leave the position at 0,0 and this method would be half-usable.  Not super convinced that that is the best idea though.

Reply:
I guess the current API choice favors `get_position` as we have `set_position` and `get_dimensions` calls